apps/aecore/lib/aecore/oracle/oracle.ex

```diff
@@ -0,0 +1,272 @@
+defmodule Aecore.Oracle.Oracle do
+  alias Aecore.Structures.OracleRegistrationTxData
+  alias Aecore.Structures.OracleQueryTxData
+  alias Aecore.Structures.OracleResponseTxData
+  alias Aecore.Structures.OracleExtendTxData
+  alias Aecore.Structures.DataTx
+  alias Aecore.Structures.SignedTx
+  alias Aecore.Txs.Pool.Worker, as: Pool
+  alias Aecore.Wallet.Worker, as: Wallet
+  alias Aecore.Chain.Worker, as: Chain
+
+  require Logger
+
+  @type ttl :: %{ttl: non_neg_integer(), type: :relative | :absolute}
+
+  @doc """
+  Registers an oracle with the given requirements for queries and responses,
+  a fee that should be paid by queries and a TTL.
+  """
+  @spec register(map(), map(), non_neg_integer(), non_neg_integer(), ttl()) :: :ok | :error
```

Comment:
be more specific about `map()` if possible for this case

Reply:
Maps should be json schemas, not possible

Reply:
but you could add a `json_schema` type to alias map for some use-cases